pkg/modulewriter/modulewriter.go

```diff
@@ -133,38 +127,92 @@ func createGroupDirs(deploymentPath string, deploymentGroups *[]config.Deploymen
 	return nil
 }
 
-func copySource(deploymentPath string, deploymentGroups *[]config.DeploymentGroup) error {
+// Get module source within deployment group
+// Rules are following:
+//   - git source
+//     => keep the same source
+//   - packer
+//     => <mod.ID>
+//   - embedded (source starts with "modules" or "comunity/modules")
+//     => ./modules/embedded/<source>
+//   - other
+//     => ./modules/<basename(source)>-<hash(abs(source))>
+func deploymentSource(mod config.Module) (string, error) {
+	if sourcereader.IsGitPath(mod.Source) {
+		return mod.Source, nil
+	}
+	switch mod.Kind {
+	case "packer":
+		return mod.ID, nil
+	case "terraform": // see below
+	default:
+		return "", fmt.Errorf("unexpected module kind %#v", mod.Kind)
+	}
 
+	if sourcereader.IsEmbeddedPath(mod.Source) {
+		return "./modules/" + filepath.Join("embedded", mod.Source), nil
+	}
+	if !sourcereader.IsLocalPath(mod.Source) {
+		return "", fmt.Errorf("unuexpected module source %s", mod.Source)
+	}
+
+	abs, err := filepath.Abs(mod.Source)
+	if err != nil {
+		return "", fmt.Errorf("failed to get absolute path for %#v: %v", mod.Source, err)
+	}
+	base := filepath.Base(mod.Source)
+	return fmt.Sprintf("./modules/%s-%s", base, shortHash(abs)), nil
+}
+
+// Returns first 4 characters of md5 sum in hex form
+func shortHash(s string) string {
+	h := md5.Sum([]byte(s))
+	return hex.EncodeToString(h[:])[:4]
+}
+
+func copyEmbeddedModules(base string) error {
+	r := sourcereader.EmbeddedSourceReader{}
+	for _, src := range []string{"modules", "community/modules"} {
+		dst := filepath.Join(base, "modules/embedded", src)
+		if err := os.MkdirAll(dst, 0755); err != nil {
+			return err
+		}
+		if err := r.CopyDir(src, dst); err != nil {
+			return err
+		}
+
+	}
+	return nil
+}
+
+func copySource(deploymentPath string, deploymentGroups *[]config.DeploymentGroup) error {
 	for iGrp := range *deploymentGroups {
 		grp := &(*deploymentGroups)[iGrp]
 		basePath := filepath.Join(deploymentPath, grp.Name)
+		if err := copyEmbeddedModules(basePath); err != nil {
```

Comment:
We copy in embedded even if they are not using any embedded modules. I think we should only do this if an embedded module is used. Thoughts?